app/assets/javascripts/discourse/app/routes/discovery-filter.js

```diff
@@ -6,11 +7,15 @@ export default class DiscoveryFilterRoute extends DiscourseRoute {
     q: { replace: true, refreshModel: true },
   };
 
-  model(data) {
-    return this.store.findFiltered("topicList", {
+  async model(data) {
+    const list = await this.store.findFiltered("topicList", {
       filter: "filter",
       params: { q: data.q },
     });
+
+    setTopicList(list);
+
+    return list;
```

Comment:
No need to change this, but destructuring allows for some nice syntax in these cases:

```suggestion
  async model({ q }) {
    const list = await this.store.findFiltered("topicList", {
      filter: "filter",
      params: { q },
    });

    setTopicList(list);

    return list;
```